src/dispatching.rs

```diff
@@ -187,12 +187,36 @@
 //! useful features. See [`examples/dispatching_features.rs`] as a more involved
 //! example.
 //!
+//! ## Dispatching or REPLs?
+//!
+//! The difference between dispatching and the REPLs ([`crate::repl`] & co) is
+//! that dispatching allows you a greater degree of flexibility at the cost of a
+//! bit more complicated setup.
+//!
+//! Here are things that dispatching can do, but REPLs can't:
+//!  - Handle different kinds of [`Update`];
+//!  - [Pass dependencies](struct.DispatcherBuilder.html#method.dependencies) to
+//!    handlers;
+//!  - Disable a [default Ctrl-C
+//!    handling](struct.DispatcherBuilder.html#method.enable_ctrlc_handler);
+//!  - Control your
+//!    [default](struct.DispatcherBuilder.html#method.default_handler) and
+//!    [error](struct.DispatcherBuilder.html#method.error_handler) handlers;
+//!  - Use [dialogues](dialogue/index.html).
+//!  - Use [`dptree`]-related functionality.
+//!  - Probably more.
```

Comment:
`;` and `.` are used inconsistently here. Maybe remove them altogether?

Reply:
Ah that's why I hate separators -- I forgot to put `;` in a few places. I think we should use just one type of separator, and the simplest one is a straight dot (`.`).

Reply:
I don't think we should use separators here at all -- the list entries are inherently separated

Reply:
In English, sentences are always to be terminated by some kind of separator (`.`, `!`, `?`, etc.), so I think we should do the same here. From `CODE_STYLE.md`:

> Be sure that your comments follow the grammar, including punctuation, the first capital letter and so on.

Reply:
It doesn't really make sense to me. Separators like dots make sense when they separate sentences from each other. But when there's other kinds of separation already (say it's the end of a `// comment`, or the end of a message bubble in a messenger, or, as in this case, the end of a list entry) the dots just add visual noise.

Reply:
A dot at the end of a book paragraph and chapter also add visual noise? People write them to stay consistent. Also, in defence of separators, they mean that the author really finished a sentence and did not just remove a final part of it due to a human factor.

> This module re-exports frequently stuff from `lorem`

> This module re-exports frequently used stuff from `lorem` and `ipsum`.

Reply:
> A dot at the end of a book paragraph and chapter also add visual noise?

Yes!

But, there is a difference, why it's a lot easier for me to tolerate dots there: paragraphs and chapters in books are a lot longer than 1-2 sentences. The more sentences you have, the less visual noise from the final `.` matters. I don't like final dots in lists like this cause the list items are very small.

Reply:
The point about human factor mistakes makes sense, though you could just as well forget/delete something and leave the dot.

Reply:
I've checked resources on the Internet, and there are a lot of punctuation style guides contradicting each other: some suggest omitting punctuation, whereas others suggest always using punctuation. It also seems that our preferred doc style is to put semicolons except for the last item. But I don't quite like this style due to mistakes caused by tracking the dot ending the last bullet.

What I suggest is always to put dots at the end of each bullet. Yes, it adds syntax noise, but at the same time, it makes some mistakes evident, like that example with "`lorem` and `ipsum`".

Reply:
I should also make a new PR since it's a separate concern, and update `CODE_STYLE.md` accordingly.

Reply:
> it makes some mistakes evident, like that example with "lorem and ipsum".

I don't agree that this is a reasonable example: as I said, you can just as well make a mistake by [not] adding `.`, so a missing `.` is not necessarily evidence for a missing part of the sentence. Besides I can't recall we made any mistakes like that in the past (even though we've made a whole bunch of other mistakes).

What I agree on is that whatever the decision on final dots in lists will be, we should make it consistent in the whole codebase and document it in `CODE_STYLE.md`.

Reply:
Okay let's leave this PR as-is then, I'll open a separate PR fixing this issue.